pkg/etcdutil/etcdutil.go

```diff
@@ -265,22 +274,40 @@ func TakeAndSaveFullSnapshot(ctx context.Context, client client.MaintenanceClose
 	timeTaken := time.Since(startTime)
 	logger.Infof("Total time taken by Snapshot API: %f seconds.", timeTaken.Seconds())
 
+	var snapshotData io.ReadCloser
+	snapshotTempDBPath := filepath.Join(tempDir, "db")
```

Comment:
Use `os.MkdirTemp` and add `defer os.RemoveAll(snapshotTempDBPath)` next line.

Reply:
the `temp` dir already exist, no need to re-create them, we just need to create a db file under this tempDir which function is doing `checkFullSnapshotIntegrity()`

Reply:
there is already a code is present to remove the `temporary: db` file which we created inside `temp` dir.
```
	defer func() {
		if err := os.Remove(snapshotTempDBPath); err != nil {
			logger.Warnf("failed to remove temporary full snapshot file: %v", err)
		}
	}()
```
we don't want to remove the `temp` dir, we just want to remove the `temporary: db` file which we created, hence `os.Remove()` is used.

Reply:
Sure, I'm fine with deleting the DB file.